tests/e2e/concatenate-xpath-strings.spec.js

```diff
@@ -0,0 +1,61 @@
+const helper = require('../helper');
+const concatenateStrings = require('../page-objects/forms/concatenate-strings.po');
+const common = require('../page-objects/common/common.po');
+const utils = require('../utils');
+const constants = require('../constants');
+
+const userContactDoc = {
+  _id: constants.USER_CONTACT_ID,
+  name: 'Jack',
+  date_of_birth: '',
+  phone: '+64274444444',
+  alternate_phone: '',
+  notes: '',
+  type: 'person',
+  reported_date: 1478469976421,
+  parent: {
+    _id: 'some_parent',
+  },
+};
+
+// If this test starts failing then we need to document in the release notes that we've removed the deprecated 
+// feature allowing for concatenation of strings
+describe('Concatenate xpath strings', () => {
+  beforeAll(done => {
+    Promise.resolve()
+      .then(() => concatenateStrings.configureForm(userContactDoc, done))
+      .catch(done.fail);
+  });
+
+  afterEach(done => {
+    utils.resetBrowser();
+    done();
+  });
+
+  afterAll(utils.afterEach);
+
+  it('concatenates strings', () => {
+    common.goToReports();
+
+    // select form
+    const addButton = element(
+      by.css('.action-container .general-actions:not(.ng-hide) .fa-plus')
+    );
+    helper.waitElementToPresent(addButton);
+    helper.clickElement(addButton);
+    element(
+      by.css(
+        '.action-container .general-actions .dropup.open .dropdown-menu li:first-child a'
```

Comment:
This seems flaky. If another form is configured and happens to be first it'll start failing. Is there another selector you can use to find the exact form you want to fill in?

Reply:
I haven't been able to find a more specific selector yet. This same selector is in multiple tests e.g. https://github.com/medic/cht-core/blob/master/tests/page-objects/forms/generic-form.po.js#L59 and https://github.com/medic/cht-core/blob/master/tests/e2e/forms/submit-photo-upload-form.spec.js#L47. Maybe we should add a new issue to update all such tests?

Reply:
Ok to can leave it as is for now.